code/modules/projectiles/guns/energy/dueling.dm

```diff
@@ -301,6 +301,8 @@
 	var/obj/projectile/energy/duel/D = loaded_projectile
 	D.setting = setting
 	D.update_appearance()
+	if(!isturf(target))
+		D.loaded_projectile.set_homing_target(target)
```

Comment:
D is already a projectile, also please get rid of one-letter var names while you're at it